### ISSUE TITLE
after the end of the unrest in the colony, a colonist “disappeared”

### ISSUE THREAD

**Issue by MrZorG33** [We-the-People-civ4col-mod/Mod#1032]
Today, when I started a new test game, I encountered a very interesting bug. 
![before unrest](https://github.com/We-the-People-civ4col-mod/Mod/assets/61798883/d80c2cfc-15ec-452b-833e-ca8105da5afb)

After the end of the unrest in the colony, a colonist “disappeared”. but at the same time, the size of the colony shows that a colonist is present in it. 
![after unrest](https://github.com/We-the-People-civ4col-mod/Mod/assets/61798883/98e10e56-ed53-4be0-a3cb-0ca1974b3ea2)

The “automatic assignment of colonists to jobs” button helps show the “invisible” colonist.
![after unrest 2](https://github.com/We-the-People-civ4col-mod/Mod/assets/61798883/78ea3f15-f053-4fed-a3f8-826d383e8a40)
[single.zip](https://github.com/user-attachments/files/16082797/single.zip)

**Comment by Razonatair** [comment]
Yeah I've noticed this before, the unrest mechanic screws with a lot of stuff, including making invisible colonists like this that are hiding somewhere until you reenable automatic to reassign them. Same thing happens in recently conquered cities sometimes.

**Comment by Nightinggale** [comment]
As I recall I  once added a workaround for this issue, which is doTurn will check for hidden units and automate them. This means if one vanishes, the problem will fix itself if you click end turn. The precise cause of how this triggered is still unknown, but it was non-trivial to figure out.

The good part of the fix is that it will also fix savegames where it had already gone wrong.

**Comment by MrZorG33** [comment]
the problem still exists. ending the turn does not solve the problem - only the automated assignment of work to colonists.
![Image](https://github.com/user-attachments/assets/efd5164d-4927-4093-bc35-4b288536f028)

[war.zip](https://github.com/user-attachments/files/18866540/war.zip)
